kliff/models/kim.py

```diff
@@ -576,19 +538,54 @@ def set_one_opt_param(self, name: str, settings: List[List[Any]]):
 
         Args:
             name: name of a fitting parameter
-            settings: initial value, flag to fix a parameter, lower and upper bounds of a
+            settings: List initial value, flag to fix a parameter, lower and upper bounds of a
                 parameter.
 
         Example:
-            name = 'param_A'
+            name = 'A'
             settings = [['default', 0, 20], [2.0, 'fix'], [2.2, 'inf', 3.3]]
             instance.set_one(name, settings)
         """
-        self.opt_params.set_one(name, settings)
+        # Keeping this API intact for now, no improvement comes to mind
```

Comment:
I agree that this function is a mess - hard to use and maintain. We will need to simplify it. Let's leave it for now and do it after the major stuff has been done.

Reply:
OK.